auth/src/main/java/org/aerogear/auth/credentials/OIDCCredentials.java

```diff
@@ -32,6 +43,68 @@ public String getRefreshToken() {
         return authState.getRefreshToken();
     }
 
+    /**
+     * A function to verify the authenticity of a JWT token against a public key, expected issuer and audience.
+     * @param jwtToken - The JWT Token to Verify.
+     * @param publicKey - The Public Key from Keycloak, without the Begin/End tags.
+     * @param issuer - The expected Issuer of the JWT
+     * @param audience - The expected Audience of the JWT
+     * @return boolean - true if the token integrity is good
+     */
+    public boolean verifyToken(String jwtToken, String publicKey, String issuer, String audience) {
+
+        // TODO Get the Public Key from the Mobile Core Config
+        // TODO Get the Audience from the Mobile Core Config
+        // TODO Construct the Issuer using properties in the Mobile Core Config
```

Comment:
@aidenkeating So the audience will just be the clientID thats in the mobile core config. 

The issuer will look like the following in the decoded JWT ` "iss": "https://keycloak.security.feedhenry.org/auth/realms/client-app"` so this will need to be constructed with something like AUTH_SERVER_URL + "/realms/" + CLIENT_ID. These can be gotten from the mobile core config too.

I think we can leave the public key as a parameter for now actually. In terms of the format, it will just be a public key without the BEGIN/END tags, as this is how keycloak displays it in the portal under. Realm Settings (Menu Item) > Keys (Tab Item) > Public Key (Button). The public key should show in a popup then. 

I think the config might be defined in this class, and this class will get the config from the mobile core config then - https://github.com/aerogear/aerogear-android-sdk/blob/33f42d7718e7abf06604ce888bdeb3f0ee2f4001/auth/src/main/java/org/aerogear/auth/AuthServiceConfig.java

Reply:
@TommyJ1994 Do we expect the user of the SDK to be invoking this function? From the original issue in Jira I'm assuming this should be hidden from the user? (https://issues.jboss.org/browse/AGDROID-686)

Reply:
@aidenkeating Originally, I think it would be of benefit for this to be called under the hood as part of the get roles functions, but its probably not guaranteed that a developer will put/have the public key in the mobile core config.

I was thinking we could maybe just expose them. We could probably do some method overloading so there are two functions:

```
private boolean verifyToken(String token) {
// get public key from the mobile core config
}

private boolean verifyToken(String token, String publicKey) {
// get public key from the param
}
```

This will allow an end user to specify a public key, rather than the one from the mobile core config.